framework/src/org/checkerframework/common/value/ValueCheckerUtils.java

```diff
@@ -104,11 +111,61 @@
         return values;
     }
 
+    public static Range getRangeFromValues(List<? extends Number> values) {
```

Comment:
I think it would again be good to ensure that the length of values is limited by MAX_VALUES.
Does this condition hold? Can you add it as additional comment here and e.g. in `getValuesFromRange` below?